pkg/subctl/cmd/gather/servicediscovery.go

```diff
@@ -110,10 +116,44 @@ func gatherConfigMapCoreDNS(info *Info) {
 	}
 }
 
+// gatherInternalServices gathers the internal service created for every exported submariner service.
+func gatherInternalServices(info *Info) {
+	services, err := info.ClientProducer.ForKubernetes().CoreV1().Services(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		info.Status.Warning("Error gathering the internal service configuration: %v", err)
+	}
```

Comment:
I think you can use a label selector that checks if the `internalSvcLabel` exists to directly list the internal services, ie

```
services, err := info.ClientProducer.ForKubernetes().CoreV1().Services(corev1.NamespaceAll).List(
    context.TODO(), metav1.ListOptions{LabelSelector:internalSvcLabel})
```

Reply:
Thanks @tpantelis, good call, ResourcesToYAMLFile is smarter than I thought at first glance. This fixed the linting collision as well  🎉

Reply:
Ahh right - `ResourcesToYAMLFile` can do it all.